src/manifest.c

```diff
@@ -812,6 +812,11 @@ static int write_manifest_plain(struct manifest *manifest)
 		}
 
 		string_or_die(&tempmanifest, "%s/Manifest.%s", manifest_tempdir, file->filename);
+		/*
+		 * use_xattr Has to match swupd-client:
+		 * there it is enabled unconditionally in verify_file().
+		 */
+		file->use_xattrs = true;
```

Comment:
Yes, it was basically broken before on systems where xattrs get set at runtime (Smack, IMA, SE Linux (?)).

Regarding the proposed fix of calling an external command on the server side: this can't be handled via some kind of external deployment, because the manifest files get created by swupd-server and then get hashed immediately. The necessary xattrs have to be set between these two steps, hence the external command.

The alternative solution is to patch swupd-client so that it does not check xattrs. I'm open for that, too, because I can't think of any good reason why checking xattrs for these data files provides additional value that would justify the complexity. But perhaps I'm just missing such a reason (additional protection against tampering when signed IMA hashes are used perhaps?).